ExecuteJobs.java

```diff
@@ -0,0 +1,80 @@
+import java.util.*;
+import java.io.*;
+
+public class ExecuteJobs {
+  public long start(String process) throws IOException {
+    // TODO some error handling
+    Process p = Runtime.getRuntime().exec(process);
+    long pid = p.pid();
+    return pid;
+  }
+
+  public int stop(long pid) throws IOException {
+    Process p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
+    BufferedReader pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+    String s = pInput.readLine();
+    if(s == null) {
+      // pid not found
+      return 2;
+    }
+
+    Runtime.getRuntime().exec("kill " + pid);
+    p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
+    pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+    s = pInput.readLine();
+    if(s != null) {
```

Comment:
if s is null here, why not inverse (return error) to reduce indentation level?

Reply:
I could do that. If s is null I could return a success error. If s is not null, I would execute everything in the if block, which would be indented one level less. If an error wasn't returned while that code executed, I would return the success error. 

I implemented the code the way I did in order to not return the same success code at two locations. I will change the implementation to your suggestion if indenting fewer levels is more important.